File: src/page/LocComp.java
package page;

import java.awt.event.MouseListener;

interface LocComp extends CustomComp {

	@Override
	public default void added(Page p) {
		getLSC().added(p);
	}

	public void addMouseListener(MouseListener ml);

	public LocSubComp getLSC();

	public String getText();

	public String getToolTipText();

	public default void reLoc() {
		getLSC().reLoc();
	}

	public default void setText(int i, String str) {
		getLSC().init(i, str);
	}

	public void setText(String t);

	public void setToolTipText(String ttt);

}

File: src/page/LocComp.kt
package page

import page.Pageimport
import java.awt.event.MouseListener

interface LocComp : CustomComp {
    override fun added(p: Page?) {
        getLSC().added(p)
    }

    fun addMouseListener(ml: MouseListener?)
    fun getLSC(): LocSubComp
    fun getText(): String?
    fun getToolTipText(): String?
    fun reLoc() {
        getLSC().reLoc()
    }

    fun setText(i: Int, str: String) {
        getLSC().init(i, str)
    }

    fun setText(t: String?)
    fun setToolTipText(ttt: String?)
}
